docs/contribute.rst

```diff
@@ -0,0 +1,268 @@
+
+Pull request
+#############
+To create a pull request on Github
+
+1. Navigate to the forked repository.
+2. Select `Compare and pull request` button.
+3. Fill in the title and comment section.
+4. Select Create pull request button.
+
+.. image:: _static/pull_request.gif
+
+Clean up
+#########
+Most of the times you need to make changes after
+pull request review so the log might look like this:
+
+.. code::
+
+    commit e8722d25cec9c03991cb30b43fdbbbcd58f66e6b
+    Author: Your Name <your.email@domain>
+    Date:   Sat Oct 13 09:47:30 2018 -0400
+
+        My second commit
+
+    commit 94d76cb8976eab9265bba8342890318a7392f383
+    Author: Your Name <your.email@domain>
+    Date:   Sat Oct 13 09:45:47 2018 -0400
+
+        My first new commit
+
+    commit 97883f921af64344fa3ac822e6492ebfd0507cdb
+    Author: Some One <Some.one@domain>
+    Date:   Sat Oct 13 09:29:58 2018 -0400
+
+        Last wifiphisher commit
+
+In this case `97883f921af64344fa3ac822e6492ebfd0507cdb` is
+the last commit before you made any changes and the ones above it are added by you.
+In order to keep the `wifiphisher` repository clean we only
+accept a single commit in a pull request. This means that you
+need to crush all these commits in to a single one.
+
+.. code:: bash
+
+    # copy the hash of the last commit before
+    # in this case 97883f921af64344fa3ac822e6492ebfd0507cdb
+
+    git rebase -i 97883f921af64344fa3ac822e6492ebfd0507cdb
+
+    # when editor opens it should look like this
+    pick 94d76cb My first new commit
+    pick e8722d2 My second commit
+
+    # change it to the following
+    pick 94d76cb My first new commit
+    f e8722d2 My second commit
+
+This will squash all those commits in to a single one
+which you can verify by running
+
+.. code:: bash
+
+    git log
+
+.. image:: _static/cleanup.gif
+
+YAPF
+#####
+You might get the following error when running test:
+
+.. code::
+
+    linters runtests: commands[0] | yapf -rpd wifiphisher
+    --- wifiphisher/pywifiphisher.py        (original)
+    +++ wifiphisher/pywifiphisher.py        (reformatted)
+    @@ -222,8 +222,7 @@
+                 should_roll_over = os.path.isfile(
+                     LOGGING_CONFIG['handlers']['file']['filename'])
+             should_roll_over and root_logger.handlers[0].doRollover()
+    -        logger.info(
+    -        "Starting Wifiphisher")
+    +        logger.info("Starting Wifiphisher")
+
+
+     def set_ip_fwd():
+    ERROR: InvocationError for command '/home/blackhatmonkey/workspace/
+    python/wifiphisher/.tox/linters/bin/yapf-rpd wifiphisher' (exited with code 1)
+
+This shows the difference between your code and
+what it should look like. You can run:
+
+.. code:: bash
+
+    yapf -i file_name
+
+YAPF will take care of the rest and automatically
+makes the required changes.
+
+It is recommended that you configure your editor
+to automatically format the code using YAPF
+to reduce manual work. For example in `visual studio code`_
```

Comment:
I'm curious. How are you using Visual Studio to run a Linux-only tool?

Reply:
It's [Visual Studio Code](https://code.visualstudio.com/) which is just a cross platform editor.